ironfish-rust/src/serializing.rs

```diff
@@ -90,40 +88,25 @@ pub(crate) mod aead {
     ///
     /// This is just a facade around the ChaCha20Poly1305 struct. It ignores
     /// nonce and aad and automatically stores the mac tag.
-    pub(crate) fn encrypt(key: &[u8], plaintext: &[u8], encrypted_output: &mut [u8]) {
-        assert_eq!(encrypted_output.len(), plaintext.len() + MAC_SIZE);
-        let mut encryptor = ChaCha20Poly1305::new(key, &[0; 8], &[0; 8]);
-        let mut tag = [0; MAC_SIZE];
-        encryptor.encrypt(
-            plaintext,
-            &mut encrypted_output[..plaintext.len()],
-            &mut tag,
-        );
-        encrypted_output[plaintext.len()..].clone_from_slice(&tag);
+    /// This is just a facade around the ChaCha20Poly1305 struct. The nonce and
+    /// associated data are zeroed.
+    pub(crate) fn encrypt(key: &[u8; 32], plaintext: &[u8]) -> Result<Vec<u8>, IronfishError> {
+        let encryptor = ChaCha20Poly1305::new(Key::from_slice(key));
+        let result = encryptor
+            .encrypt(&Nonce::default(), plaintext)
```

Comment:
Want to verify that using `&Nonce::default()` is acceptable.

Reply:
The key is a one-time-use key generated for this particular note/output, so this should be acceptable.